### DISCUSSION TITLE
Control button single click in normal mode.

### DISCUSSION THREAD

**Discussion by planevina** [HomeSpan/HomeSpan#521] [Ideas]
The control pin function is below now:

normal mode :  
long click --> enter command mode;
single click --> not use

command mode:
long click --> select menu;
single click --> change menu;

Can homespan throw a HS_STATUS message like 'HS_NORMAL_SINGLE_CLICK'  in normal mode
Then user can use it do some function himself ?

**Comment by HomeSpan** [comment]
Since single-click is not being used for the control button when in "normal" mode, , adding in a callback would not interfere with the command menu.  I can add this in the next release, as well as a callback for double-clicking the control button (since that's not used either).  Thanks for the suggestion.

**Reply by HomeSpan** [reply]
I should have reviewed my code before replying above.  Single-click is actually _already used_ by the control button.  It's set to trigger after 3 seconds, which is why it seems to be a long press.  But long press is also used --- it is set to 10 seconds and causes a factory reset.  Unfortunately, adding in single-click callback won't be possible.

**Reply by planevina** [reply]
OK ,thanks for reply . In fact I tried to add it ,and found that it's conflict with "enter command mode" .

**Reply by planevina** [reply]
I will try to remove "factory reset(it's not useful for me, or move it to command mode)" and add single click again ^^
